File: src/main/resources/templates/cpp/tree/MultiNode.cc
// Copyright 2024 HiveVM.ORG. All rights reserved.
// Copyright (c) 2006, Sun Microsystems, Inc. All rights reserved.
// SPDX-License-Identifier: BSD-3-Clause
//
// Derived from JavaCC 7.0.12: src/main/resources/templates/cpp/MultiNode.cc.template, src/main/resources/templates/cpp/SimpleNode.cc.template

// disable warnings on parser header files
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-parameter"
#pragma GCC diagnostic ignored "-Wunused-variable"

#include "__NODE_TYPE__.h"
//@if(VISITOR)
#include "__PARSER_NAME__Visitor.h"
//@fi

//@if(CPP_NAMESPACE)
namespace __CPP_NAMESPACE__ {
//@fi
	
__NODE_TYPE__::__NODE_TYPE__(int id) : __NODE_CLASS__(id) {
}
__NODE_TYPE__::~__NODE_TYPE__() {
}

//@if(VISITOR)
/** Accept the visitor. **/
__VISITOR_RETURN_TYPE__ __NODE_TYPE__::jjtAccept(__PARSER_NAME__Visitor *visitor, __VISITOR_DATA_TYPE__ data) const {
//@if(!VISITOR_RETURN_TYPE_VOID)
	return
//@fi
	visitor->visit(this, data);
}
//@fi

//@if(CPP_NAMESPACE)
}
//@fi

#pragma GCC diagnostic pop

File: src/main/resources/templates/java/tree/MultiNode.java
// Copyright 2024 HiveVM.ORG. All rights reserved.
// Copyright (c) 2006, Sun Microsystems, Inc. All rights reserved.
// SPDX-License-Identifier: BSD-3-Clause
//
// Derived from JavaCC 7.0.12: org/javacc/jjtree/ASTOptions.java, src/main/resources/templates/MultiNode.template

package __JAVA_PACKAGE__;

public class __NODE_TYPE__ extends __NODE_CLASS__ {

	public __NODE_TYPE__(Parser p, int id) {
		super(p, id);
	}
//@if(VISITOR)

//@if(VISITOR_EXCEPTION)
	public __VISITOR_RETURN_TYPE__ jjtAccept(NodeVisitor visitor, __VISITOR_DATA_TYPE__ data) throws __VISITOR_EXCEPTION__ {
//@else
	public __VISITOR_RETURN_TYPE__ jjtAccept(NodeVisitor visitor, __VISITOR_DATA_TYPE__ data) {
//@fi
//@if(VISITOR_RETURN_TYPE_VOID)
		visitor.visit(this, data);
//@else
		return visitor.visit(this, data);
//@fi
	}
//@fi
//@if(NODE_FACTORY)

	public static Node jjtCreate(Parser p, int id) {
		return new __NODE_TYPE__(p, id);
	}
//@fi
}